x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/LuceneSlice.java

```diff
@@ -19,14 +24,39 @@ public record LuceneSlice(
     boolean queryHead,
     ShardContext shardContext,
     List<PartialLeafReaderContext> leaves,
-    Weight weight,
+    Query query,
+    ScoreMode scoreMode,
     List<Object> tags
 ) {
+
     int numLeaves() {
         return leaves.size();
     }
 
     PartialLeafReaderContext getLeaf(int index) {
         return leaves.get(index);
     }
+
+    Weight createWeight() {
+        var searcher = shardContext.searcher();
+        try {
+            Weight w = searcher.createWeight(query, scoreMode, 1);
+            return new OwningWeight(query, w);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static class OwningWeight extends FilterWeight {
```

Comment:
Could we put the `Query` into `LuceneScorer` instead of into the `Weight`? There's already a bunch of useful stuff. We could do:
```
+        private final Query query;
         private final Weight weight;
```

and check there.

Reply:
Sure, I've updated the LuceneScorer to keep both the query and tags.